File: Python/String/Largest Number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        # li = [str(num) for num in nums]
        li = []
        for num in nums:
            li.append(str(num))
        def compare(x, y):
        # Custom comparator to decide the order
            if x + y > y + x:
                return -1
            elif x + y < y + x:
                return 1
            else:
                return 0
        li.sort(key=cmp_to_key(compare))
        if li[0]=="0":
            return "0"
        return "".join(li)

File: CPP/String/Largest Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string>st;
        for(auto num:nums){
            st.push_back(to_string(num));
        }
        sort(st.begin(),st.end(),[](string &a, string &b){
            return a+b>b+a;
        });
        if(st[0]=="0")return "0";
        string larg;
        for(auto i:st){
            larg=larg+i;
        }
        return larg;
    }
};
